src/Queue.js

```diff
@@ -60,21 +62,38 @@ class Queue {
     // If it isn't finished, emit a scheduler interrupt notifying the scheduler that this process
     // needs to be moved to a lower priority queue
     manageTimeSlice(currentProcess, time) {
+        if (currentProcess.isStateChanged()) {
+            this.quantumClock = 0;
+            return;
+        } else {
+            this.quantumClock += time;
```

Comment:
This is another example where you could have just left the whole else statement off and put the line on it's own after the if statement 
`this.quantumClock += time;`
but again this 100% acceptable just showing an example of a different way (so to speak).